src/main/java/org/opensearch/geospatial/index/query/xyshape/XYShapeQueryProcessor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.query.xyshape;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import org.apache.lucene.document.ShapeField;
+import org.apache.lucene.document.XYShape;
+import org.apache.lucene.geo.XYGeometry;
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.opensearch.common.geo.ShapeRelation;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.GeometryVisitor;
+import org.opensearch.geospatial.index.mapper.xyshape.XYShapeFieldMapper;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.QueryShardException;
+
+/**
+ * Query Processor to convert given Geometry into Lucene query
+ */
+public class XYShapeQueryProcessor {
```

Comment:
does this processor needs to extends/implement some interface? How this processor and which function of this processor is getting called to create the query?

Reply:
Yes. It will be called by field type. That will be my next PR